lua/litee/symboltree/init.lua

```diff
@@ -338,6 +338,21 @@ function M.dump_node()
     lib_tree.dump_tree(ctx.node)
 end
 
+local select_config_key = function(key, user_config, default_config)
+  local val = user_config[key]
+  local default_val = config[key]
+
+  if "table" ~= type(default_val) then
+    return val
+  end
+
+  if val == nil then
+    return nil
+  end
+
+  return vim.tbl_extend("keep", val, default_val)
```

Comment:
I'm a little confused about this "tbl_extend" why is this called with two values ?

Reply:
When, for example, the user needs to override only one custom mapping, it needs to make only one entry in `keymaps`. The rest remain the default.

Reply:
And this work okay when both values are *not* a map?

Reply:
In this place both values are tables. Note that the function returns earlier otherwise. Please find a bug if I'm wrong.

Reply:
Okay got it. looks good